pygpu/gpuarray.pyx

```diff
@@ -1465,6 +1465,26 @@ cdef class GpuArray:
         else:
             raise IndexError, "cannot index with: %s" % (key,)
 
+    def write(self, np.ndarray src not None):
+        cdef size_t npsz = np.PyArray_NBYTES(src)
+        cdef size_t sz = gpuarray_get_elsize(self.ga.typecode)
+        cdef unsigned i
+        for i in range(self.ga.nd):
+            sz *= self.ga.dimensions[i]
+        if sz != npsz:
+            raise ValueError, "GpuArray and Numpy array do not have the same size in bytes"
+        array_write(self, np.PyArray_DATA(src), sz)
```

Comment:
This will read out of bounds if the source is not contiguous.  You will also get strange results if the source and destination don't agree on strides.

Finally this will definitely give weird results if the data types don't agree.

I thinks the checks should be extended to cover those.